source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -459,18 +459,16 @@ bool Filter::maybeTunnel(const std::string& cluster_name) {
     if (!cluster) {
       return false;
     }
-    // TODO(snowp): Ideally we should prevent this from being configured, but that's tricky to get
-    // right since whether a cluster is invalid depends on both the tcp_proxy config + cluster
-    // config.
-    if ((cluster->info()->features() & Upstream::ClusterInfo::Features::HTTP2) == 0) {
-      ENVOY_LOG(error, "Attempted to tunnel over HTTP/1.1, this is not supported. Set "
-                       "http2_protocol_options on the cluster.");
-      return false;
-    }
 
```

Comment:
Huh, I just realized should be getting the HTTP connection pool from the cluster manager rather than creating one every time.  I'll go ahead and fix that once this lands, but can I ask you to write an HTTP/1 "no reuse" test (or tests) where we 

CONNECT, downstream close, new CONNECT request
and
CONNECT, upstream close, new CONNECT request

and make sure both the new CONNECT requests land on new upstream connections?  Kind of like the H2 connection reuse test but the reverse.

Reply:
Let me know if this is not what you meant: https://github.com/envoyproxy/envoy/pull/13293/commits/2e0857f5435ecd53c30d58c502fc5054808c3721